src/Stratis.Bitcoin.Tests/Consensus/ConsensusTestExtensions.cs

```diff
@@ -40,5 +40,24 @@ public static ChainedHeader[] ToArray(this ChainedHeader chainedHeader, int head
 
             return headers;
         }
+
+        public static bool HaveBlockDataAvailabilityStateOf(this ConnectNewHeadersResult connectNewHeadersResult, BlockDataAvailabilityState blockDataAvailabilityState)
+        {
+            if ((connectNewHeadersResult.DownloadFrom == null) || (connectNewHeadersResult.DownloadTo == null))
+            {
+                return false;
+            }
+
+            ChainedHeader chainedHeader = connectNewHeadersResult.DownloadTo;
+            while (chainedHeader.Height >= connectNewHeadersResult.DownloadFrom.Height)
+            {
+                if (chainedHeader.BlockDataAvailability != blockDataAvailabilityState)
+                {
+                    return false;
+                }
+                chainedHeader = chainedHeader.Previous;
+            }
+            return true;
+        }
```

Comment:
i was thinking more like this:

```
        public static bool HaveBlockDataAvailabilityStateOf(this ConnectNewHeadersResult connectNewHeadersResult, BlockDataAvailabilityState blockDataAvailabilityState)
        {
            if ((connectNewHeadersResult.DownloadFrom == null) || (connectNewHeadersResult.DownloadTo == null))
            {
                return false;
            }

            ChainedHeader chainedHeader = connectNewHeadersResult.DownloadTo;
            while (chainedHeader.Height >= connectNewHeadersResult.DownloadFrom.Height)
            {
                if (chainedHeader.BlockDataAvailability != blockDataAvailabilityState)
                {
                    return false;
                }

                chainedHeader = chainedHeader.Previous;
            }

            return true;
        }
```

that is in compliance with the code style. Another option would be
```
        public static bool HaveBlockDataAvailabilityStateOf(this ConnectNewHeadersResult connectNewHeadersResult, BlockDataAvailabilityState blockDataAvailabilityState)
        {
            if ((connectNewHeadersResult.DownloadFrom == null) || (connectNewHeadersResult.DownloadTo == null))
                return false;

            ChainedHeader chainedHeader = connectNewHeadersResult.DownloadTo;
            while (chainedHeader.Height >= connectNewHeadersResult.DownloadFrom.Height)
            {
                if (chainedHeader.BlockDataAvailability != blockDataAvailabilityState)
                    return false;
   
                chainedHeader = chainedHeader.Previous;
            }

            return true;
        }
```

Reply:
All good - used the former.